src/main/java/com/minecolonies/coremod/inventory/ContainerRack.java

```diff
@@ -129,7 +138,7 @@ public ContainerRack(@NotNull final TileEntityRack tileEntityRack, @Nullable fin
                         playerInventory,
                         j + i * INVENTORY_COLUMNS + INVENTORY_COLUMNS,
                         PLAYER_INVENTORY_INITIAL_X_OFFSET + j * PLAYER_INVENTORY_OFFSET_EACH,
-                        PLAYER_INVENTORY_INITIAL_Y_OFFSET + PLAYER_INVENTORY_OFFSET_EACH * this.inventorySize + i * PLAYER_INVENTORY_OFFSET_EACH
+                        PLAYER_INVENTORY_INITIAL_Y_OFFSET + extraOffset + PLAYER_INVENTORY_OFFSET_EACH * Math.min(this.inventorySize, 8) + i * PLAYER_INVENTORY_OFFSET_EACH
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 8 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)